xmtp_id/src/verifier.rs

```diff
@@ -0,0 +1,47 @@
+use ethers::types::{Address, BlockId, Signature};
+use ethers::providers::{Http, Middleware, Provider};
+use std::convert::TryFrom;
+use std::sync::Arc;
+use crate::bindings::coinbase_smart_wallet::CoinbaseSmartWallet;
+use anyhow::Error;
+
+const BASE_SEPOLIA_RPC_URL: &str = "https://sepolia.base.org";
+const EIP1271_MAGIC_VALUE: [u8; 4] = [0x16, 0x26, 0xba, 0x7e];
+
+pub struct ERC1271Verifier {
```

Comment:
I'm not sure if this is possible, but could we just import the ERC1271 Interface Smart contract, and verify the signature with that? In this case, that might support all ERC1271 signatures and not just Coinbase Smart Wallet.

Otherwise, I'm in favor of renaming ERC-1271 to `CoinbaseSmartWallet`, and implementing the CredentialVerifier trait on that. This could also mean we would have to support each smart wallet individually

Reply:
Each ERC-1271 wallet(Gnosis, Coinbase and others I don't know of) has its own verification logic so an ERC-1271 interface doesn't suffice and we will need to support each SCW individually. 
Renaming makes sense to me.

Reply:
Do we need to worry about generating the signatures, or will that be something the SDK  does?

AFAIU every smart wallet will implement `is_valid_signature`, so even if the signature generation or signing is different, signature verification will always be that function which coincides with `ERC-1271`. The only difference will be the address that `is_valid_signature` will be called on since each SCW will implement its own verification, but it must be according to ERC-1271

Reply:
Yeah that makes sense. 

I was trying to figure out the whole lifecycle from message to signature but here I think you are right we just need to put a simple ERC-1271 `is_valid_signature` interface here and all additional logic should be used in the tests.

Reply:
Seconding that. At verification time we want this to be generic for any valid ERC-1271. The only things that should be implementation-specific are test-cases, where we would want to create specific Smart Wallets to test against.